### ISSUE TITLE
URL init will ignore the \t in query string

### ISSUE THREAD

**Issue by tadelesh** [aio-libs/yarl#861]
### Describe the bug

![image](https://user-images.githubusercontent.com/1726438/234870749-a85e7928-d1c6-4645-97f0-d829c0de72ea.png)

### To Reproduce

1. Create URL with `\t` in query string
2. The parsed URL ignores the `\t`

### Expected behavior

The URL should parse the `\t` 

### Logs/tracebacks

```python-traceback
No log
```

### Python Version

```console
Python 3.10.10
```

### multidict Version

```console
Name: multidict
Version: 6.0.4
Summary: multidict implementation
Home-page: https://github.com/aio-libs/multidict
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: d:\workspace\tadelesh\python-issue-test\.venv\lib\site-packages
Requires:
Required-by: aiohttp, yarl
```

### yarl Version

```console
Name: yarl
Version: 1.9.2
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl/
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache-2.0
Location: d:\workspace\tadelesh\python-issue-test\.venv\lib\site-packages
Requires: idna, multidict
Required-by: aiohttp
```

### OS

Windows

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

**Comment by mjpieters** [comment]
A minimal reproducible test case:

```python
from yarl import URL

url = URL("http://example.com/path?param=value\twith\ttabs")
assert '\t' in url.query["param"]
```

The tab characters have been stripped. The same applies to `\n` newlines.

**Comment by mjpieters** [comment]
The behaviour comes from the Python standard library:

```
>>> from urllib.parse import urlsplit
>>> urlsplit('http://example.com/path?param=value\twith\ttabs').query
'param=valuewithtabs'
```

`urlsplit()` treats `\t`, `\r` and `\n` as unsafe bytes, as per the [WHATWG URL specification](https://url.spec.whatwg.org/#concept-basic-url-parser). This behaviour is *by design*, and should not be overridden.

Sorry, that makes this a wontfix.

**Comment by webknjaz** [comment]
@mjpieters wouldn't hurt to have a test still, just to explicitly document the expectation, right?

**Comment by mjpieters** [comment]
@webknjaz perhaps, want to tackle that?
